src/formats/adfformat.cpp

```diff
@@ -354,6 +355,284 @@ namespace OpenBabel {
+    // NOTREADABLE  READONEONLY  NOTWRITABLE  WRITEONEONLY
+    virtual unsigned int Flags()
+    {
+      return READONEONLY | NOTWRITABLE;
+    };
+
+    /// The "API" interface functions
+    virtual bool ReadMolecule(OBBase* pOb, OBConversion* pConv);
+  };
+  //***
+
+  //Make an instance of the format class
+  ADFDftbFormat theADFDftbFormat;
+
+  /////////////////////////////////////////////////////////////////
+  bool ADFDftbFormat::ReadMolecule(OBBase* pOb, OBConversion* pConv)
+  {
+    OBMol* pmol = pOb->CastAndClear<OBMol>();
+    if (!pmol)
+      return false;
+
+    //Define some references so we can use the old parameter names
+    istream &ifs = *pConv->GetInStream();
+    OBMol &mol = *pmol;
+    const char* title = pConv->GetTitle();
+
+    char buffer[BUFF_SIZE];
+    vector<string> vs;
+
+    mol.BeginModify();
+
+    while (ifs.getline(buffer, BUFF_SIZE)) {
+      if (strcmp(buffer, "Geometry") == 0) {
+        // We need to clear the atoms before proceeding. Since this comes
+        // before all the other data, we can just clear the whole molecule
+        mol.Clear();
+        mol.BeginModify();
+
+        ifs.getline(buffer, BUFF_SIZE); // ------------
+        ifs.getline(buffer, BUFF_SIZE); // Atoms
+        ifs.getline(buffer, BUFF_SIZE);
+
+        // Make sure it is the correct line
+        if (strstr(buffer, "Index") && strstr(buffer, "Symbol")) {
+          double lengthConversion = 1.0;
+          // Check the units
+          if (strstr(buffer, "bohr"))
+            lengthConversion = BOHR_TO_ANGSTROM;
+
+          while (ifs.getline(buffer, BUFF_SIZE)) {
+            tokenize(vs, buffer);
+            // Should be of size 5
+            if (vs.size() < 5)
+              break;
+
+            OBAtom* atom = mol.NewAtom();
+            atom->SetAtomicNum(OBElements::GetAtomicNum(vs[1].c_str()));
+            double x = atof(vs[2].c_str()) * lengthConversion;
+            double y = atof(vs[3].c_str()) * lengthConversion;
+            double z = atof(vs[4].c_str()) * lengthConversion;
+            atom->SetVector(x, y, z);
+          }
+        }
+
+        // Now read the lattice vectors
+        ifs.getline(buffer, BUFF_SIZE);
+        if (strstr(buffer, "Lattice vectors")) {
+          double lengthConversion = 1.0;
+          // Check the units
+          if (strstr(buffer, "bohr"))
+            lengthConversion = BOHR_TO_ANGSTROM;
+
+          std::vector<vector3> vectors;
+          for (short i = 0; i < 3; ++i) {
+            ifs.getline(buffer, BUFF_SIZE);
+            tokenize(vs, buffer);
+            if (vs.size() != 4)
+              break;
+
+            double x = atof(vs[1].c_str()) * lengthConversion;
+            double y = atof(vs[2].c_str()) * lengthConversion;
+            double z = atof(vs[3].c_str()) * lengthConversion;
+            vectors.push_back(vector3(x, y, z));
+          }
+
+          while (vectors.size() < 3)
+            vectors.push_back(vector3(0.0, 0.0, 0.0));
```

Comment:
Same here - is there any possibility of a 1D or 2D calculation?

Reply:
Same response as above.